src/bootstrap/compile.rs

```diff
@@ -1206,41 +1206,13 @@ pub fn run_cargo(builder: &Builder<'_>,
         deps.push((path_to_add.into(), false));
     }
 
-    // Now we want to update the contents of the stamp file, if necessary. First
```

Comment:
I'm actually sort of surprised that more of this method wasn't deleted. We're still executing Cargo in JSON mode and dealing with all that jazz, but is that to track more esoteric aspects? Could we perhaps prune most of the infrastructure here to avoid calling Cargo in JSON mode?

Reply:
We need to know where the artifacts rustc has emitted are to copy them into the sysroot (and some other things); I'm not sure there's a way to do that without Cargo running in JSON mode?

Reply:
👍